python/lsst/cp/verify/verifyPtc.py

```diff
@@ -0,0 +1,211 @@
+            # than a certain threshold (default: 5%).
+            verify['GAIN'] = bool(diffGain < self.config.gainThreshold)
+            verify['NOISE'] = bool(diffNoise < self.config.noiseThreshold)
+
+            # Check that the measured PTC turnoff is at least greater than the
+            # full-well requirement of 90k e-.
+            turnoffCut = self.config.turnoffThreshold
+            verify['PTC_TURNOFF'] = bool(calib.ptcTurnoff[ampName]*calib.gain[ampName] > turnoffCut)
+            # Check the a00 value (brighter-fatter effect).
+            # This is a purely electrostatic parameter that should not change
+            # unless voltages are changed (e.g., parallel, bias voltages).
+            # Check that the fitted a00 parameter per CCD vendor is within a
+            # range motivated by measurements on data (DM-30171).
+            if ptcFitType in ['EXPAPPROXIMATION', 'FULLCOVARIANCE']:
+                # a00 is a fit parameter from these models.
+                if ptcFitType == 'EXPAPPROXIMATION':
+                    a00 = calib.ptcFitPars[ampName][0]
+                else:
+                    a00 = calib.aMatrix[ampName][0][0]
+                if detVendor == 'ITL':
+                    a00Max = self.config.a00MaxITL
+                    a00Min = self.config.a00MinITL
+                    verify['BFE_A00'] = bool(a00 > a00Min and a00 < a00Max)
+                elif detVendor == 'E2V':
+                    a00Max = self.config.a00MaxE2V
+                    a00Min = self.config.a00MinE2V
+                    verify['BFE_A00'] = bool(a00 > a00Min and a00 < a00Max)
+                else:
+                    raise RuntimeError(f"Detector type {detVendor} not one of 'ITL' or 'E2V'")
+            # Overall success among all tests for this amp.
+            verify['SUCCESS'] = bool(np.all(list(verify.values())))
+            if verify['SUCCESS'] is False:
+                success = False
+
+            verifyStats[ampName] = verify
+
+        # Loop over amps to make a detector summary.
+        verifyDetStats = {'GAIN': [], 'NOISE': [], 'PTC_TURNOFF': [], 'BFE_A00': []}
+        for amp in verifyStats:
+            for testName in verifyStats[amp]:
+                if testName == 'SUCCESS':
+                    continue
+                verifyDetStats[testName].append(verifyStats[amp][testName])
+
+        # If ptc model did not fit for a00 (e.g., POLYNOMIAL)
+        if not len(verifyDetStats['BFE_A00']):
+            verifyDetStats.pop('BFE_A00')
+
+        # Overwrite verifyDetStats with final boolean test over all amps
+        verifyDetStatsFinal = {}
+        for testName in verifyDetStats:
+            testBool = bool(np.all(list(verifyDetStats[testName])))
+            # Save the tests that failed
+            if not testBool:
+                verifyDetStatsFinal[testName] = bool(np.all(list(verifyDetStats[testName])))
+
+        return verifyDetStatsFinal, bool(success)
```

Comment:
Github is complaining about the cases of the booleans in the VerifyDefect.yaml pipeline definition, but doesn't let me put a comment there, so it's here instead.